metric/internal/global/state_test.go

```diff
@@ -31,13 +29,15 @@ func resetGlobalMeterProvider() {
 func TestSetMeterProvider(t *testing.T) {
 	t.Cleanup(resetGlobalMeterProvider)
 
-	t.Run("Set With default panics", func(t *testing.T) {
+	t.Run("Set With default is a noop", func(t *testing.T) {
 		resetGlobalMeterProvider()
+		SetMeterProvider(MeterProvider())
 
-		assert.Panics(t, func() {
-			SetMeterProvider(MeterProvider())
-		})
-
+		_, ok := MeterProvider().(*meterProvider)
+		if !ok {
+			t.Error("Global Meter Provider should be the default meter provider")
+			return
```

Comment:
I see that `t.Fatal` is used in the codebase so I would rather refactor it to the following for sake of consistency:

```suggestion
			t.Fatal("Global Meter Provider should be the default meter provider")
```

same comment applies to other places

Reply:
Done (I didn't apply your suggestion, in order to do it everywhere).